src/main/java/uk/gov/hmcts/darts/arm/component/ArmRetentionEventDateCalculator.java

```diff
@@ -0,0 +1,7 @@
+package uk.gov.hmcts.darts.arm.component;
+
+public interface ArmRetentionEventDateCalculator {
+
+    void calculateRetentionEventDate(Integer externalObjectDirectoryId);
```

Comment:
Could return a boolean to signify a failure state. Be nice if clients could react to a failure by putting out a log message

Reply:
There are logs for any failures. Not sure its relevant for the automated task.